src/test_creation/modules/code_analyzer/repo.py

```diff
@@ -25,7 +36,56 @@ def __init__(self, path: str):
             '.c': 'C'
         }
         self.lf_map = self._get_language_file_map()
+        self.ffl_map = self._get_file_function_lineno_map()
+        try:
+            self._get_git_metadata()
+        except Exception as e:
+            logger.info(e)
+
+    def _get_git_metadata(self):
+        config = ConfigParser()
+        if os.path.exists(self.path + '/.git/config'):
+            config.read(self.path + '/.git/config')
+        else:
+            raise FileNotFoundError('/.git/config does not exist')
+
+        self.url = config['remote "origin"']['url']
+        
+        if 'git@' in self.url:
+            self.mode = 'ssh'
+            pattern = 'git@(.*?):(.*?)/(.*?).git'
+        elif 'https' in self.url:
+            self.mode = 'https'
+            pattern = 'https://(.*?)/(.*?)/(.*?).git'
+            
+        self.service, self.user, self.name = re.search(pattern, self.url).group(1,2,3)
+
+        if 'branch "master"' in list(config):
+            self.main_branch = 'master'
+        elif 'branch "main"' in list(config):
+            self.main_branch = 'main'
 
+        return {
+            'mode': self.mode,
+            'service': self.service,
+            'user': self.user,
+            'name': self.name,
+            'main_branch': self.main_branch
+        }
+
+    def _get_git_direct_link(self, file: str, lineno: int = None):
+        link = f'https://{self.service}/{self.user}/{self.name}/blob/{self.main_branch}/{file}'
+        if lineno:
+            link += f'#L{lineno}'
+        return link
+
+    def _get_relative_path(self, file: str):
+        path = file.replace(self.path, '', 1)
+        if path[0] == '/':
+            return path
+        else:
+            return '/' + path
```

Comment:
Path starting with `/` cannot be considered as relative path. Consider to all `./` instead of `/`.

Also, consider to use [`os.path.relpath`](https://docs.python.org/3/library/os.path.html#os.path.relpath) for getting relative path.

Reply:
Maybe relative path is not a good name.
The purpose of this function is to get a path so that I could append it to the base link, e.g. "github.com/lyst/lightfm/blob/master"

Reply:
fixed in [7f02ed5](https://github.com/UBC-MDS/test-creation/pull/107/commits/7f02ed5d156bd50c905616c3b85aec8b885fbac0)